core/src/main/scala/chisel3/Data.scala

```diff
@@ -372,6 +372,89 @@ trait BaseType extends HasId with NamedComponent {
     */
   private[chisel3] def cloneTypeFull: this.type
 
+  // Internal API: returns a ref that can be assigned to, if consistent with the binding
+  private[chisel3] def lref: Node = {
+    requireIsHardware(this)
+    requireVisible()
+    topBindingOpt match {
+      case Some(binding: ReadOnlyBinding) =>
+        throwException(s"internal error: attempted to generate LHS ref to ReadOnlyBinding $binding")
+      case Some(ViewBinding(target)) => reify(target).lref
+      case Some(binding: TopBinding) => Node(this)
+      case opt => throwException(s"internal error: unknown binding $opt in generating LHS ref")
+    }
+  }
+
+  // Internal API: returns a ref, if bound
+  private[chisel3] final def ref: Arg = {
+    def materializeWire(makeConst: Boolean = false): Arg = {
+      if (!Builder.currentModule.isDefined) throwException(s"internal error: cannot materialize ref for $this")
+      implicit val sourceInfo = UnlocatableSourceInfo
+      this match {
+        case (data: Data) => {
+          if (makeConst) {
+            WireDefault(Const(chiselTypeOf(data)), data).ref
+          } else {
+            WireDefault(data).ref
+          }
+        }
```

Comment:
```suggestion
        case data: Data =>
          if (makeConst) {
            WireDefault(Const(chiselTypeOf(data)), data).ref
          } else {
            WireDefault(data).ref
          }
```
As a general coding style, you don't need the parentheses nor the curly braces. It's a stylistic nit but we generally don't include them in this codebase so probably best to be consistent.